packages/core/src/stateUtils.ts

```diff
@@ -1688,7 +1705,7 @@ export function macrostep(
       internalQueue
     );
     shouldSelectEventlessTransitions = nextSnapshot !== previousState;
-    states.push(nextSnapshot);
+    addMicrostate(nextSnapshot, nextEvent, enabledTransitions);
```

Comment:
it would be great to somehow explicitly differentiate between raised microstates and the "always" microstates - that's a useful information for the end user

Reply:
We could have `event: null` in the inspection event (matches SCXML's concept of "null events")

Reply:
Yeah, I was thinking about a similar thing. It would be great to know what microsteps were produced by what transitions - but if we wont be able to figure out how to "link" them on the inspector side of things then we'll just add more metadata to those events.

Reply:
I realized that we already differentiate - we can see what kind of `eventType` caused the transition on `inspectionEvent.transitions[0].eventType`. This is actually more useful than `event: null` or specifying whether an event is from an always-transition or not because the transition object is more specific about that (can be an always event, or a wildcard, or a partial wildcard...)